saleor/core/utils/__init__.py

```diff
@@ -79,7 +79,7 @@ def build_absolute_uri(location, is_secure=False):
     except AttributeError:
         raise ImproperlyConfigured('You need to specify CANONICAL_HOSTNAME in '
                                    'your Django settings file')
-    if not absolute_http_url_re.match(location):
+    if not re.compile(r"^https?://", re.I).match(location):
```

Comment:
But what is the point of compiling the regex at this point? It's just as slow as `re.match(…)` as you recompile each time the condition is checked.